src/Export/ExportItemAdapter.php

```diff
@@ -35,26 +44,52 @@ class ExportItemAdapter implements ExportItemAdapterInterface
     /** @var ExportContext */
     protected $exportContext;
 
+    /** @var LoggerInterface  */
+    private $logger;
+
     public function __construct(
         ContainerInterface $container,
         RouterInterface $router,
         EventDispatcherInterface $eventDispatcher,
         Config $config,
         AdapterFactory $adapterFactory,
-        ExportContext $exportContext
+        ExportContext $exportContext,
+        LoggerInterface $logger
     ) {
         $this->container = $container;
         $this->router = $router;
         $this->eventDispatcher = $eventDispatcher;
         $this->config = $config;
         $this->adapterFactory = $adapterFactory;
         $this->exportContext = $exportContext;
+        $this->logger = $logger;
     }
 
-    public function adapt(Item $item, ProductEntity $product): Item
+    public function adapt(Item $item, ProductEntity $product): ?Item
     {
         $this->eventDispatcher->dispatch(new BeforeItemAdaptEvent($product, $item), BeforeItemAdaptEvent::NAME);
 
+        try {
+            $item = $this->adaptProduct($item, $product);
+        } catch (EmptyValueNotAllowedException | ProductInvalidException | Exception | Throwable $exception) {
```

Comment:
Shouldn't `Throwable` catch all exceptions anyway? I think the other types are unnecessary.